amtl/am-string.h

```diff
@@ -140,6 +156,14 @@ class AString
     chars_[length] = '\0';
   }
 
+  static AString concat(const char *a, size_t alen, const char *b, size_t blen) {
+    AutoArray<char> buf;
+    buf = new char[alen + blen + 1];
+    strcpy(buf, a);
+    strcat(buf, b);
```

Comment:
You are right, `strcpy` is more efficent. I didn't understand about wrapping, so i used `IsUintPtrAddSafe`